The focus of this challenge is: 
* Choose a more descriptive title for the bar chart
* Redo the class analysis with data from a more recent month/year

# **Notebook for exploring data from the Brazilian Health Ministry repository**

[Link for data extraction](http://tabnet.datasus.gov.br/cgi/tabcgi.exe?sih/cnv/qiSP.def)

**Access date and time:** 20th may, 2021 at 22h20

**Information to access data:**
* **Localização:** Informações de Saúde(TABNET)/Assistência à Saúde/Dados Consolidados AIH (RD), por local de internação, a partir de 2008/Brasil por município de São Paulo
* **Linha:** Município
* **Coluna:** Ano/mes processamento
* **Conteúdo:** Valor Total
* **Período:** de Jan-2008 à Mar-2021


## **1. Importing software libraries**


In [9]:
import pandas as pd

import numpy as np

import seaborn as sns

import matplotlib
import matplotlib.pyplot as plt
import matplotlib.dates as mdates
import matplotlib.ticker as ticker

## **2. Importing and reading data**


In [10]:
url_dados = 'https://github.com/SaraZachi/BootCamp-2021/blob/main/Aula%201/Data/A001458189_28_143_208.csv?raw=true'
dados = pd.read_csv(url_dados, encoding='ISO-8859-1', engine='python',
                    skiprows=3, skipfooter=12, sep=';', 
                    thousands='.', decimal=',')
dados

Município     2008/Jan  ...      2021/Mar          Total
0              350010 Adamantina    192360.05  ...     843298.96    65988020.00
1                   350030 Aguaí     58020.37  ...             -     4672764.57
2        350050 Águas de Lindóia     36102.19  ...      30613.53     8013242.34
3                  350070 Agudos     41090.60  ...      53310.41     9783863.36
4             350100 Altinópolis     22628.13  ...      33494.84     4781172.12
..                           ...          ...  ...           ...            ...
355             355680 Viradouro     27566.36  ...      13067.88     3327839.11
356  355690 Vista Alegre do Alto      1638.60  ...             -       87560.45
357            355700 Votorantim    130982.55  ...     596052.93    30858652.62
358           355710 Votuporanga    200699.61  ...    1559836.72   153939697.35
359                        Total  91678473.48  ...  392946147.42 41396604304.13

[360 rows x 161 columns]

2.1. Analysing dataframe (DF)

*   The df documented 359 municipalities + Lables, giving us 360 rows
*   From Jan/2008 to Mar/2021, we have 13 years * 12 months (hole years) + 3 months (from Jan-Mar/2021), which gives us 159 columns + 2 for lable amd city name, giving us 161 columns

Therfore, our dataframe has 360 rows × 161 columns


In [11]:
dados.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 360 entries, 0 to 359
Columns: 161 entries, Município to Total
dtypes: float64(1), object(160)
memory usage: 452.9+ KB


## **3. Formatting data**

3.1. As default or data was presented in scientific notation. In order to change that, we use the following codes to present them with two numbers decimals.

In [12]:
pd.options.display.float_format = '{:.2f}'.format
dados['2008/Abr'].mean

<bound method Series.mean of 0         304813.18
1          57386.91
2          42259.47
3          79860.83
4          23629.02
           ...     
355        31829.34
356         2583.86
357       165420.20
358       413376.22
359    161227125.43
Name: 2008/Abr, Length: 360, dtype: object>

3.2. In order to plot a bar chart with the amount of money spent by São Paulo with the Brazilian public health plan during December 2020, the blank spots in the data frame '2020/Dez' column had to be substituted with a number (0). These 'blank spots' are identified with '-'. Therefore, the following code was applied.

In [15]:
dados['2020/Dez'] = dados['2020/Dez'].replace('-', 0.0)
dados['2020/Dez'] = dados['2020/Dez'].astype(float)

In [16]:
dados['2020/Dez']

0        464135.09
1             0.00
2         51543.40
3         30959.27
4         21347.98
          ...     
355       25942.46
356           0.00
357      432882.75
358     1338367.09
359   330489045.98
Name: 2020/Dez, Length: 360, dtype: float64

3.3. The main data frame has way too many cities to fit in a single bar chart, therefore, only the top 10 on the list will be used for this project.

In [ ]:
cod_municipios = dados['Município'].head(10)
cod_municipios

In [ ]:
data = dados.query('Município in @cod_municipios')
data

In [ ]:
xticks = ['350010 Adamantina', '350030 Aguaí', '350050 Águas de Lindóia0', '350070 Agudos', '350100 Altinópolis', '350110 Alto Alegre', 
          '350130 Álvares Machado', ' 350160 Americana', '350170 Américo Brasiliense', '350190 Amparo']
xticks

## **04. Plotting a bar chart**

In [ ]:
sns.set()
ax = data.sort_values(by = '2020/Dez', ascending=False).plot(x='Município', y='2020/Dez', kind='bar', 
                figsize=(10,6),
                color='#fdbf11')

ax.yaxis.set_major_formatter(ticker.StrMethodFormatter('{x:,.2f}'))

plt.title('Amount spent on public medical care in 10 cities of São Paulo on December 2020', fontsize=16, 
          fontweight='bold', 
          color='#000000')

plt.yticks(fontsize=10)
plt.xticks(fontsize=10)

ax.set_xlabel('Municipalitie', fontsize=14, fontweight='bold', color='#000000')
ax.set_ylabel('Amount spent (R$)', fontsize=14, fontweight='bold', color='#000000')

plt.show()